File: contact_relation.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# 关系类型定义
RELATION_TYPES = {
    'TEACHER': '师生',
    'COLLEAGUE': '同事',
    'SUPERIOR': '上下级',
    'FRIEND': '朋友',
    'FAMILY': '家人',
    'CLASSMATE': '同学',
    'OTHER': '其他'
}

def determine_relation(contact):
    """根据联系人信息判断关系"""
    name = contact.get('name', '')
    notes = contact.get('notes', '').lower()
    relation = contact.get('relation', '')

    if relation:
        return relation

    if '老师' in notes or '导师' in notes:
        return RELATION_TYPES['TEACHER']
    if '同事' in notes or '工作' in notes:
        return RELATION_TYPES['COLLEAGUE']
    if '领导' in notes or '上司' in notes or '下属' in notes:
        return RELATION_TYPES['SUPERIOR']
    if '朋友' in notes or '好友' in notes:
        return RELATION_TYPES['FRIEND']
    if '家人' in notes or '父母' in notes or '子女' in notes or '兄弟姐妹' in notes or '配偶' in notes:
        return RELATION_TYPES['FAMILY']
    if '同学' in notes or '校友' in notes:
        return RELATION_TYPES['CLASSMATE']

    return RELATION_TYPES['OTHER']

def get_title_by_relation(relation, name):
    """根据关系获取称谓"""
    if relation == RELATION_TYPES['TEACHER']:
        return name + '老师' if '老师' not in name else name
    if relation == RELATION_TYPES['SUPERIOR']:
        return name + '总' if '总' not in name and '经理' not in name else name
    if relation == RELATION_TYPES['FAMILY']:
        return name
    return name

File: contactRelation.js
// 联系人关系判断模块

// 关系类型定义
const RELATION_TYPES = {
  TEACHER: '师生',
  COLLEAGUE: '同事',
  SUPERIOR: '上下级',
  FRIEND: '朋友',
  FAMILY: '家人',
  CLASSMATE: '同学',
  OTHER: '其他'
};

// 根据联系人信息判断关系
function determineRelation(contact) {
  const { name, notes, relation } = contact;

  // 如果已经有明确的关系标签，直接返回
  if (relation) {
    return relation;
  }

  // 根据备注信息判断关系
  const lowerNotes = (notes || '').toLowerCase();

  if (lowerNotes.includes('老师') || lowerNotes.includes('导师')) {
    return RELATION_TYPES.TEACHER;
  }

  if (lowerNotes.includes('同事') || lowerNotes.includes('工作')) {
    return RELATION_TYPES.COLLEAGUE;
  }

  if (lowerNotes.includes('领导') || lowerNotes.includes('上司') || lowerNotes.includes('下属')) {
    return RELATION_TYPES.SUPERIOR;
  }

  if (lowerNotes.includes('朋友') || lowerNotes.includes('好友')) {
    return RELATION_TYPES.FRIEND;
  }

  if (lowerNotes.includes('家人') || lowerNotes.includes('父母') || lowerNotes.includes('兄弟姐妹') ||
      lowerNotes.includes('配偶') || lowerNotes.includes('子女')) {
    return RELATION_TYPES.FAMILY;
  }

  if (lowerNotes.includes('同学') || lowerNotes.includes('校友')) {
    return RELATION_TYPES.CLASSMATE;
  }

  return RELATION_TYPES.OTHER;
}

// 获取关系对应的称谓
function getTitleByRelation(relation, name) {
  switch (relation) {
    case RELATION_TYPES.TEACHER:
      return name.includes('老师') ? name : name + '老师';
    case RELATION_TYPES.SUPERIOR:
      return name.includes('总') || name.includes('经理') ? name : name + '总';
    case RELATION_TYPES.FAMILY:
      return name; // 家人通常直接称呼名字或昵称
    case RELATION_TYPES.CLASSMATE:
    case RELATION_TYPES.FRIEND:
    case RELATION_TYPES.COLLEAGUE:
    case RELATION_TYPES.OTHER:
    default:
      return name;
  }
}

module.exports = {
  RELATION_TYPES,
  determineRelation,
  getTitleByRelation
};